include/envoy/network/connection_handler.h

```diff
@@ -68,9 +68,50 @@ class ConnectionHandler {
    * after they have been temporarily disabled.
    */
   virtual void enableListeners() PURE;
+
+  /**
+   * Used by ConnectionHandler to manage listeners.
+   */
+  class ActiveListener {
+  public:
+    virtual ~ActiveListener() = default;
+
+    /**
+     * @return the tag value as configured.
+     */
+    virtual uint64_t listenerTag() PURE;
+    /**
+     * @return the actual Listener object.
+     */
+    virtual ListenerPtr& listener() PURE;
+  };
+
+  using ActiveListenerPtr = std::unique_ptr<ActiveListener>;
 };
 
 using ConnectionHandlerPtr = std::unique_ptr<ConnectionHandler>;
 
+/**
+ * A registered factory interface to create different kinds of
+ * ActiveUdpListener.
+ */
+class ActiveUdpListenerFactory {
```

Comment:
Q: To future proof this, should we just call this `ActiveListenerFactory`? I'm wondering if maybe later we will have other listener types. I don't feel strongly about this, just throwing it out there.

Reply:
I don't think we want another TCP active listener impl along with ActiveTcpListener, do we?

Reply:
Yeah it's fine to leave as is for now.